fg/apps/factory/views/twist.py

```diff
@@ -369,7 +369,7 @@ def import_plate_task(rows, fields, factory_order):
 
                    # Look for a sample already associated with plate from the factory order      
                     for plate_well in plate.wells.all():
-                        contenders = Sample.objects.filter(wells=plate_well)
+                        contenders = Sample.objects.filter(wells=plate_well).filter(part=part)
                         if contenders.count() > 0:
```

Comment:
okay great! So you've tested this locally and it works as you would like?

Reply:
Yes I have tested locally and works as I would like